circuits/cpp/barretenberg/cpp/src/barretenberg/srs/factories/file_crs_factory.hpp

```diff
@@ -20,38 +49,41 @@ template <typename Curve> class FileProverCrs : public ProverCrs<Curve> {
     std::shared_ptr<typename Curve::AffineElement[]> monomials_;
 };
 
-class FileVerifierCrs : public VerifierCrs {
+template <typename Curve> class FileVerifierCrs : public VerifierCrs<Curve> {
   public:
-    FileVerifierCrs(std::string const& path);
-
-    ~FileVerifierCrs();
+    FileVerifierCrs(std::string const& path, const size_t num_points);
+};
 
-    g2::affine_element get_g2x() const override;
+template <> class FileVerifierCrs<curve::BN254> : public VerifierCrs<curve::BN254> {
+    using Curve = curve::BN254;
 
-    pairing::miller_lines const* get_precomputed_g2_lines() const override;
+  public:
+    FileVerifierCrs(std::string const& path, const size_t num_points = 0);
+    virtual ~FileVerifierCrs();
+    Curve::G2AffineElement get_g2x() const override { return g2_x; };
+    pairing::miller_lines const* get_precomputed_g2_lines() const override { return precomputed_g2_lines; };
+    Curve::AffineElement get_first_g1() const override { return first_g1; };
```

Comment:
write comment to describe what get_first_g1 is and why we need it

Reply:
and for miller lines

Reply:
I added comments in the `crs_factory.hpp` for these two function given that's the base class the other factories derive from. They are high level as I don't think it's the right place to go into detail on Miller's algorithm on Shplonk but let me know if you think otherwise.